src/internal/components/chart-series-details/index.tsx

```diff
@@ -1,49 +1,117 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React, { memo } from 'react';
+import React, { memo, ReactNode, useEffect, useRef } from 'react';
 import clsx from 'clsx';
 
 import { BaseComponentProps, getBaseProps } from '../../base-component';
-import InternalBox from '../../../box/internal';
-import { ChartDetailPair } from '../../../pie-chart/interfaces';
 import ChartSeriesMarker, { ChartSeriesMarkerType } from '../chart-series-marker';
 import styles from './styles.css.js';
+import InternalExpandableSection from '../../../expandable-section/internal';
+import getSeriesDetailsText from './series-details-text';
+
+interface ChartDetailPair {
+  key: ReactNode;
+  value: ReactNode;
+}
 
 export interface ChartSeriesDetailItem extends ChartDetailPair {
   markerType?: ChartSeriesMarkerType;
   color?: string;
   isDimmed?: boolean;
+  subItems?: ReadonlyArray<ChartDetailPair>;
+  expandable?: boolean;
 }
+export type ExpandedStates = Record<number, boolean>;
 
 export interface ChartSeriesDetailsProps extends BaseComponentProps {
   details: ReadonlyArray<ChartSeriesDetailItem>;
+  expandedStates?: ExpandedStates;
```

Comment:
This has to be defined outside because we want to keep the expanded state of each expandable series of each X coordinate independently, and otherwise the state would be lost when closing and therefore unmounting the popover. And it needs to be optional because this component is used by all charts but the feature is not supported by area or pie charts.